File: MbUnit/Saucery/Driver/SauceryRemoteWebDriver.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Remote;

namespace Saucery.Driver {
    public class SauceryRemoteWebDriver : RemoteWebDriver {
        public SauceryRemoteWebDriver(ICommandExecutor commandExecutor, ICapabilities desiredCapabilities)
            : base(commandExecutor, desiredCapabilities) {
        }

        public SauceryRemoteWebDriver(Uri remoteAddress, ICapabilities desiredCapabilities)
            : base(remoteAddress, desiredCapabilities, TimeSpan.FromSeconds(400)) {
            //var threadLocalDriver = new ThreadLocal<RemoteWebDriver>();
            //threadLocalDriver.Value(ThreadGuard.protect(new RemoteWebDriver(remoteAddress, desiredCapabilities)));
            //return threadLocalDriver.Value;
        }

        public String GetSessionId() {
            return SessionId != null ? SessionId.ToString() : "BLANK";
            //return SessionId.ToString();
        } 
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 * 
 */

File: JUnit/saucery-core/src/main/java/au/com/fullcirclesolutions/saucery/driver/SauceryRemoteWebDriver.java
package au.com.fullcirclesolutions.saucery.driver;

import java.net.URL;

import org.openqa.selenium.remote.CommandExecutor;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.remote.RemoteWebDriver;

public class SauceryRemoteWebDriver extends RemoteWebDriver {

    public SauceryRemoteWebDriver(CommandExecutor commandExecutor, DesiredCapabilities desiredCapabilities) {
        super(commandExecutor, desiredCapabilities);
    }

    public SauceryRemoteWebDriver(URL remoteAddress, DesiredCapabilities desiredCapabilities) {
        super(remoteAddress, desiredCapabilities);
    }

    public String GetSessionId() {
        return getSessionId().toString();
    }
}
/*
 * Copyright Andrew Gray, Full Circle Solutions
 * Date: 12th July 2014
 */
